.github/workflows/tests.yaml

```diff
@@ -18,7 +18,7 @@ jobs:
         uses: actions/checkout@v3
       - uses: actions/setup-go@v4
         with:
-          go-version: '~1.20.3'
```

Comment:
btw since we are using `~` it is not necessary to change minor versions. `'~1.20.3'` will use anything greater or equal to `1.20.3`
https://devhints.io/semver

Reply:
Yes, in this context, it is true. Nevertheless, it shouldn't do any harm to increase the minimum version allowed.

Excluding golang and only focusing on other packages' versions, years of experience have proven that you shall be better off locking your dependencies and not using `~`. Breaking changes can sneak in patch releases, when you'll expect it the least. Thus, having strict control can be preferable compared to letting it looser. You'll spare some debugging hours at the expense of not being always up-to-date!